### PR TITLE
Blog App Data Model

### PR DESCRIPTION
**Pull request by AminaBuhari**
Create a new app. Give it a meaningful name.
    Initialize your project with Git.
    Make sure that your project has Postgres database set up.
    Build your project schema.
    Create and run the necessary migration files.
    Table and column names should match the ERD diagram.
    note: photo for users table should be a link to an image
    Foreign keys should be included.
    All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Invalid Code Review Request ❌

Hi @AminaBuhari, This is @sja-thedude your code reviewer.

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

You have submitted a project, that doesn't have linters. Kindly set up linters correctly and make sure all have green check marks.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: db/migrate/20220709110049_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+      t.integer :comments_counter
+      t.integer :likes_counter
+      t.integer :author_id
```
Comment (aimemalaika) [inline db/migrate/20220709110049_create_posts.rb:8]: - [ ] author_id should be a foreign key from the user table, Kindly reference it

File: db/migrate/20220709110140_create_likes.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.integer :post_id
+      t.integer :author_id
```
Comment (aimemalaika) [inline db/migrate/20220709110140_create_likes.rb:5]: - [ ] Kindly reference these two columns to their primary keys in their respective tables

File: db/migrate/20220709110201_create_comments.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.integer :post_id
+      t.integer :author_id
```
Comment (aimemalaika) [inline db/migrate/20220709110201_create_comments.rb:5]: - [ ] Kindly apply apply the previous comment given on the likes migration on the comments

File: db/migrate/20220709112249_add_user_to_post.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddUserToPost < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```
Comment (aimemalaika) [inline db/migrate/20220709112249_add_user_to_post.rb:3]: - [ ] Kindly use the correct indexes given in the ERD 'author_id'

**Review by aimemalaika — CHANGES_REQUESTED** [review]
Hi @AminaBuhari, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220709110049_create_posts.rb (hunk 1 above)
Comment (AminaBuhari) [reply db/migrate/20220709110049_create_posts.rb:8]: user_id actually. i've updated it appropriately. Thank you for the input.

File: db/migrate/20220709112249_add_user_to_post.rb (hunk 4 above)
Comment (AminaBuhari) [reply db/migrate/20220709112249_add_user_to_post.rb:3]: Hi i referenced user_id because its the user table that's created. Which has been updated.

**Review by dasileker — APPROVED** [review]
### **Hi @AminaBuhari,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**